src/emulator/auth/operations.ts

```diff
@@ -944,12 +941,23 @@ export function setAccountInfoImpl(
 
     if (reqBody.email) {
       assert(isValidEmailAddress(reqBody.email), "INVALID_EMAIL");
+      if (!emulatorUrl) {
+        throw new Error("Internal assertion error: Emulator URL invalid");
+      }
+
       const newEmail = canonicalizeEmailAddress(reqBody.email);
       if (newEmail !== user.email) {
         assert(!state.getUserByEmail(newEmail), "EMAIL_EXISTS");
         updates.email = newEmail;
         // TODO: Set verified if email is verified by IDP linked to account.
         updates.emailVerified = false;
+        if (user.email && !user.initialEmail) {
+          updates.initialEmail = user.email;
+        }
+        const initialEmail = user.initialEmail ?? updates.initialEmail;
+        if (initialEmail && newEmail !== initialEmail) {
+          sendOobForEmailReset(state, initialEmail, emulatorUrl);
```

Comment:
You may want to move this side effect to the very end, after `updateUserByLocalId` to minimize the chance of out-of-order issues with developers' test code.

Reply:
Done